code/math/staticrand.cpp

```diff
@@ -20,8 +20,9 @@ void init_semirand()
 {
 	Semirand_inited = true;
 
+	// Originally this made a 30-bit rand by sticking two 15-bit rands from myrand() together. Instead we trim rand32() down to size.
 	for (auto & number : Semirand)
-		number = (static_cast<unsigned>(myrand()) << 15u) + myrand();
+		number = (unsigned) (rand32() & 0x3fffffff);
```

Comment:
I think you should use STATIC_RAND_MAX here instead of the number itself.

Reply:
Good point; I hadn't defined the constant when I wrote that. Fixed.